File: python/53_maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_current = max_global = nums[0]

        if len(nums) == 1:
            return max_global

        for i in range(1, len(nums)):
            max_current = max(max_current + nums[i], nums[i])
            if max_global < max_current:
                max_global = max_current

        return max_global

File: go/53_maximum-subarray.go
func maxSubArray(nums []int) int {
	maxCurrent := nums[0]
	maxGlobal := maxCurrent

	for _, n := range nums[1:] {
		maxCurrent = maxInt(maxCurrent+n, n)
		maxGlobal = maxInt(max_Global, max_Current)
	}

	return maxGlobal
}

func maxInt(a, b int) int {
	if a > b {
		return a
	} else {
		return b
	}
}